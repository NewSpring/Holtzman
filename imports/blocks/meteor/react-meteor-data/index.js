
import createContainer from "./createContainer";

export {
  createContainer,
};
