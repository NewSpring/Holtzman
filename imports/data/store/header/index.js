/*
  Share action types
*/
import reducer from "./reducer";
import "./saga";
import { addReducer } from "../utilities";

addReducer({
  header: reducer,
});

export default {
  show: (options = {}) => ({
    type: "HEADER.TOGGLE_VISIBILITY",
    visible: typeof options.visible === "undefined" ? true : options.visible,
    statusBar: options.statusBar,
  }),
  hide: (options = {}) => ({
    type: "HEADER.TOGGLE_VISIBILITY",
    visible: typeof options.visible === "undefined" ? false : options.visible,
    statusBar: options.statusBar,
  }),

  set: (content, requestee) => ({ type: "HEADER.SET", content, requestee }),
  color: (color) => ({ type: "HEADER.SET", content: { color } }),
  title: (title) => ({ type: "HEADER.SET", content: { title } }),
  light: (light) => ({ type: "HEADER.SET", content: { light } }),
  isSearch: (isSearch, searchSubmit) => ({
    type: "HEADER.SET",
    content: { isSearch, searchSubmit },
  }),

  lock: (requestee) => ({ type: "HEADER.LOCK", requestee }),
  unlock: () => ({ type: "HEADER.UNLOCK" }),

  // XXX this currently doesn't affect the store
  statusBarColor: (color) => ({ type: "STATUSBAR.SET", color }),
};
