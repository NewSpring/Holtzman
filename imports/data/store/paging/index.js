import reducer from "./reducer";
import { addReducer } from "../utilities";

addReducer({
  paging: reducer,
});

export default {
  increment: () => ({ type: "PAGING.INCREMENT" }),
  pause: () => ({ type: "PAGING.PAUSE" }),
  resume: () => ({ type: "PAGING.RESUME" }),
  reset: () => ({ type: "PAGING.RESET" }),
};
