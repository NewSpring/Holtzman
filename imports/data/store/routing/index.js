
// import "./saga"

// Constants
// taken from https://github.com/reactjs/react-router-redux/blob/master/src/index.js
const TRANSITION = "@@router/TRANSITION";
const UPDATE_LOCATION = "@@router/UPDATE_LOCATION";

const SELECT_LOCATION = (state) => state.routing.location;

function transition(method) {
  return (...args) => ({
    type: TRANSITION,
    payload: { method, args },
  });
}

export const push = transition("push");
export const replace = transition("replace");
export const go = transition("go");
export const goBack = transition("goBack");
export const goForward = transition("goForward");

export const routeActions = { push, replace, go, goBack, goForward };

function updateLocation(location) {
  return {
    type: UPDATE_LOCATION,
    payload: location,
  };
}

// Reducer

const initialState = {
  location: undefined,
};

export function routeReducer(state = initialState, { type, payload: location }) {
  if (type !== UPDATE_LOCATION) {
    return state;
  }

  return { ...state, location };
}

// Syncing

export function syncHistory(history) {
  let unsubscribeHistory;
  let currentKey;
  let unsubscribeStore;

  let connected = false;
  let syncing = false;

  history.listen((location) => { initialState.location = location; })();

  function middleware(store) {
    unsubscribeHistory = history.listen((location) => {
      currentKey = location.key;
      if (syncing) {
        // Don't dispatch a new action if we're replaying location.
        return;
      }
      const { routing } = store.getState();
      if (!location.previous) {
        location.previous = []; // eslint-disable-line no-param-reassign
      }
      if (!routing.location.previous) {
        routing.location.previous = [];
      }

      if (routing.location.previous[routing.location.previous.length - 1] === location.pathname) {
        routing.location.previous.splice(-1);
        location.previous = routing.location.previous; // eslint-disable-line no-param-reassign
      } else {
        // eslint-disable-next-line no-param-reassign
        location.previous = [...routing.location.previous, ...[routing.location.pathname]];
      }


      store.dispatch(updateLocation(location));
    });

    connected = true;

    // eslint-disable-next-line consistent-return
    return (next) => (action) => {
      if (action.type !== TRANSITION || !connected) {
        return next(action);
      }

      const { payload: { method, args } } = action;

      history[method](...args);
    };
  }

  middleware.listenForReplays =
    (store, selectLocationState = SELECT_LOCATION) => {
      const getLocationState = () => selectLocationState(store.getState());
      const initialLocation = getLocationState();

      unsubscribeStore = store.subscribe(() => {
        const location = getLocationState();

        // If we're resetting to the beginning, use the saved initial value. We
        // need to dispatch a new action at this point to populate the store
        // appropriately.
        if (location.key === initialLocation.key) {
          history.replace(initialLocation);
          return;
        }

        // Otherwise, if we need to update the history location, do so without
        // dispatching a new action, as we're just bringing history in sync
        // with the store.
        if (location.key !== currentKey) {
          syncing = true;
          history.transitionTo(location);
          syncing = false;
        }
      });
    };

  middleware.unsubscribe = () => {
    unsubscribeHistory();
    if (unsubscribeStore) {
      unsubscribeStore();
    }

    connected = false;
  };

  return middleware;
}
