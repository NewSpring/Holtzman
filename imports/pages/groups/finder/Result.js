/* eslint-disable max-len */
import { Component, PropTypes } from "react";
import { connect } from "react-redux";
import { graphql } from "react-apollo";
import { withRouter } from "react-router";
import gql from "graphql-tag";

import { nav as navActions } from "../../../store";
// import ReactTooltip from "react-tooltip";
// import Truncate from "truncate";

import Split, { Left, Right } from "../../../blocks/split";
import GoogleMap from "../../../components/map";
import Layout from "./ResultLayout";

// HACK
let internalIp = null;
if (Meteor.isClient) {
  // NOTE: window.RTCPeerConnection is "not a constructor" in FF22/23
  const RTCPeerConnection = window.RTCPeerConnection ||
    window.mozRTCPeerConnection ||
    window.webkitRTCPeerConnection;   // compatibility for firefox and chrome

  if (RTCPeerConnection) {
    const pc = new RTCPeerConnection({ iceServers: [] });
    const noop = () => {};
    // create a bogus data channel
    pc.createDataChannel("");
    // create offer and set local description
    pc.createOffer(pc.setLocalDescription.bind(pc), noop);
    // listen for candidate events
    pc.onicecandidate = (ice) => {
      if (!ice || !ice.candidate || !ice.candidate.candidate) return;
      const myIP = /([0-9]{1,3}(\.[0-9]{1,3}){3}|[a-f0-9]{1,4}(:[a-f0-9]{1,4}){7})/
        .exec(ice.candidate.candidate)[1];
      internalIp = myIP;
      pc.onicecandidate = noop;
    };
  }
}

const mapStateToProps = ({ routing: { location } }) => {
  const tags = Object.keys(location.query).length && location.query.tags ? location.query.tags : "";
  const q = Object.keys(location.query).length && location.query.q ? location.query.q : null;
  const campuses = (
    Object.keys(location.query).length && location.query.campuses ? location.query.campuses : ""
  );
  return { tags, q, location, campuses };
};

const CAMPUS_LOCATION_QUERY = gql`
  query GetCampuses { campuses { entityId, id, name } }
`;

const withCampusLocations = graphql(CAMPUS_LOCATION_QUERY, { name: "campusLocations" });

const GROUP_FINDER_QUERY = gql`
  query GroupFinder($query: String, $tags: [String], $limit: Int, $offset: Int, $ip: String, $campuses: [String]) {
    groups(query: $query, attributes: $tags, limit: $limit, offset: $offset, clientIp: $ip, campuses: $campuses) {
      count
      results {
        id
        name
        entityId
        type
        kidFriendly
        demographic
        description
        photo
        ageRange
        distance
        schedule { description }
        locations { location { latitude, longitude } }
        tags { id, value }
        campus { name, entityId }
      }
    }
  }
`;

const withGroupFinder = graphql(GROUP_FINDER_QUERY, {
  options: ownProps => ({
    ssr: false,
    variables: {
      tags: ownProps.tags.split(",").filter(x => x),
      query: ownProps.q,
      ip: internalIp,
      limit: 10,
      offset: 0,
      campuses: ownProps.campuses.split(",").filter(x => x),
    },
  }),
  props: ({ data }) => ({
    data,
    paginate: () => data.fetchMore({
      variables: { offset: data.groups.results.length },
      updateQuery: (previousResult, { fetchMoreResult }) => {
        if (!fetchMoreResult.data) { return previousResult; }
        return {
          groups: {
            count: fetchMoreResult.data.groups.count,
            // Append the new feed results to the old one
            results: [...previousResult.groups.results, ...fetchMoreResult.data.groups.results],
          },
        };
      },
    }),
  }),
});

const defaultArray = [];
@withRouter
@withCampusLocations // enables this query to be static
@connect(mapStateToProps)
@withGroupFinder
export default class Template extends Component {

  static propTypes = {
    q: PropTypes.string,
    dispatch: PropTypes.func.isRequired,
    tags: PropTypes.string.isRequired,
    location: PropTypes.object.isRequired,
    router: PropTypes.object.isRequired,
    /* eslint-disable */
    data: PropTypes.shape({
      loading: PropTypes.bool,
      groups: PropTypes.shape({
        count: PropTypes.number,
        results: PropTypes.array,
      }),
    }),
    /* eslint-enable */
    campusLocations: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
    campuses: PropTypes.string.isRequired,
    paginate: PropTypes.func.isRequired,
  }

  state = {
    showTags: false,
    showSearch: false,
    hover: null,
  }

  componentWillMount() {
    if (this.props.q) this.setState({ showSearch: true });
    this.props.dispatch(navActions.setLevel("BASIC_CONTENT"));
  }

  onCardHover = (e) => {
    const { id } = e.currentTarget;
    this.setState({ hover: id });
  }

  onMarkerHover =marker => this.setState({ hover: marker.id })

  getMarkers = (groups = []) => {
    const markers = [];
    return _.uniq(markers.concat(groups
      .filter(x => x.locations && x.locations.length && x.locations[0].location)
      .map(x => ({
        latitude: x.locations[0].location.latitude,
        longitude: x.locations[0].location.longitude,
        id: x.id,
        // children: this.createChild(x),
      }))
      .filter(x => x.latitude && x.longitude)
    ), x => x.id);
  }

  toggleSearch = () => this.setState({ showSearch: !this.state.showSearch })
  toggleTags = () => this.setState({ showTags: !this.state.showTags })

  removeQueryString = (e) => {
    if (e) e.preventDefault();
    const { location, router } = this.props;

    if (location && location.query && location.query.q) delete location.query.q;
    const newPath = router.createPath(location);
    router.replace(newPath);
  }

  render() {
    const { data, tags, campusLocations, campuses, q } = this.props;
    let count;
    let groups = defaultArray;
    if (data.groups && data.groups.count) count = data.groups.count;
    if (data.groups && data.groups.results) groups = data.groups.results;

    let isMobile;
    if (typeof window !== "undefined" && window !== null) {
      isMobile = window.matchMedia("(max-width: 768px)").matches;
    }

    const markers = this.getMarkers(groups);

    return (
      <div>
        <Split>
          {/* Map */}
          <Right mobile={false} classes={["background--left"]}>
            {(() => {
              if (isMobile || Meteor.isServer) return null;
              return (
                <GoogleMap
                  autoCenter
                  markers={markers}
                  onMarkerHover={this.onMarkerHover}
                  hover={this.state.hover}
                  onChildClick={({ id }) => this.props.router.push(`/groups/${id}`)}
                />
              );
            })()}
          </Right>
        </Split>
        <Left scroll classes={["background--light-secondary"]}>
          <Layout
            loading={data.loading}
            groups={groups}
            count={count}
            tags={tags && tags.split(",").filter(x => x)}
            campuses={campuses && campuses.split(",").filter(x => x)}
            campusLocations={campusLocations}
            query={q}
            removeQueryString={this.removeQueryString}
            showTags={this.state.showTags}
            toggleTags={this.toggleTags}
            showSearch={this.state.showSearch}
            toggleSearch={this.toggleSearch}
            onCardHover={this.onCardHover}
            paginate={this.props.paginate}
          />
        </Left>
      </div>
    );
  }
}
