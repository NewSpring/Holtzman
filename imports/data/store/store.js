import "regenerator-runtime/runtime";

import { createStore, combineReducers, compose, applyMiddleware } from "redux";
import { ApolloProvider } from "react-apollo";
import createSagaMiddleware from "redux-saga";
import reduxReset from "redux-reset";

import GraphQL from "../graphql";

import { reducers, middlewares, sagas } from "./utilities";
import { syncHistory, routeReducer } from "./routing";

const createReduxStore = (initialState, history) => {
  if (initialState) {
    // bug with SSR
    delete initialState.nav; // eslint-disable-line no-param-reassign
  }

  const joinedReducers = { ...reducers,
    ...{
      routing: routeReducer,
      apollo: GraphQL.reducer(),
    },
  };

  const sharedMiddlewares = [...middlewares, GraphQL.middleware()];

  const reduxRouterMiddleware = syncHistory(history);

  const sagaMiddleware = createSagaMiddleware();

  let sharedCompose = [
    applyMiddleware(
      ...sharedMiddlewares,
      sagaMiddleware,
      reduxRouterMiddleware,
    ),
    reduxReset(),
  ];

  if (process.env.NODE_ENV !== "production") {
    sharedCompose = [...sharedCompose, ...[
      typeof window === "object" && typeof window.devToolsExtension !== "undefined" ?
        window.devToolsExtension() :
        f => f,
    ]];
  }

  const store = compose(...sharedCompose)(createStore)(
    combineReducers(joinedReducers), initialState,
  );

  sagas.forEach(saga => sagaMiddleware.run(saga()));

  return store;
};


const wrapper = ApolloProvider;

export {
  wrapper,
  createReduxStore,
};
