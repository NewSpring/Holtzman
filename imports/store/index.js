
import accounts from "./accounts";
import audio from "./audio";
import give from "./give";
import liked from "./liked";
import live from "./live";
import modal from "./modal";
import nav from "./nav";
import paging from "./paging";
import responsive from "./responsive";
import routing from "./routing";
import search from "./search";
import sections from "./sections";
import share from "./share";
import topics from "./topics";
// import fullscreen from "./fullscreen";

let header;
if (process.env.NATIVE) header = require("./header").default;

import { wrapper, createReduxStore } from "./store";
import { addMiddleware, addReducer, createReducer, addSaga } from "./utilities";

export {
  accounts,
  audio,
  give,
  header,
  liked,
  live,
  modal,
  nav,
  paging,
  responsive,
  routing,
  search,
  sections,
  share,
  topics,

  wrapper,
  createReduxStore,

  addMiddleware,

  addReducer,
  createReducer,
};
