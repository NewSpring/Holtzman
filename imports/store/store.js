import "regenerator-runtime/runtime";

import { Component, PropTypes } from "react";
import { createStore, combineReducers, compose, applyMiddleware } from "redux";
import { ApolloProvider } from "react-apollo";
import createSagaMiddleware from "redux-saga";
import reduxReset from "redux-reset";

import { GraphQL } from "../graphql";

import { reducers, middlewares, sagas } from "./utilities";
import { syncHistory, routeReducer } from "./routing";

const createReduxStore = (initialState, history) => {

  if (initialState) {
    // bug with SSR
    delete initialState.nav;
  }

  const joinedReducers = {...reducers, ...{
    routing: routeReducer,
    apollo: GraphQL.reducer(),
  }};

  let sharedMiddlewares = [...middlewares, ...GraphQL.middleware()];

  const reduxRouterMiddleware = syncHistory(history);

  const sagaMiddleware = createSagaMiddleware();
  let sharedCompose = [
    applyMiddleware(
      ...sharedMiddlewares,
      sagaMiddleware,
      reduxRouterMiddleware
    ),
    reduxReset(),
  ];

  if (process.env.NODE_ENV != "production") {
    sharedCompose = [...sharedCompose, ...[
      typeof window === "object" && typeof window.devToolsExtension !== "undefined" ? window.devToolsExtension() : f => f
    ]];
  }

  const store = compose(...sharedCompose)(createStore)(
    combineReducers(joinedReducers), initialState
  );

  sagas.forEach(saga => sagaMiddleware.run(saga()));

  return store;

};


const wrapper = ApolloProvider;

export {
  wrapper,
  createReduxStore
};
