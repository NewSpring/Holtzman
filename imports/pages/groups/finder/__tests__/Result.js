import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { Meteor } from "meteor/meteor";
import { nav as navActions } from "../../../../store";
import { TemplateWithoutData as Template } from "../Result";

jest.mock("../../../../store", () => ({
  nav: {
    setLevel: jest.fn(),
  },
}));

const defaultProps = {
  q: "test",
  dispatch: jest.fn(),
  tags: "one,two",
  location: {},
  router: {
    createPath: jest.fn(),
    replace: jest.fn(),
  },
  done: false,
  loading: false,
  Loading: jest.fn(),
  data: {
    groups: {
      count: 2,
      results: [
        {
          id: "1",
          locations: [
            {
              location: {
                latitude: 2,
                longitude: 3,
              },
            },
          ],
        },
        {
          id: "2",
          locations: [
            {
              location: {
                latitude: 2,
                longitude: 3,
              },
            },
          ],
        },
      ],
    },
  },
  campusLocations: [],
  campuses: "one,two",
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return <Template { ...newProps } />;
};

window.matchMedia = jest.fn().mockReturnValue({
  matches: false,
});

it("renders with props", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders without map on mobile", () => {
  const mockMatchMedia = jest.fn().mockReturnValue({
    matches: true,
  });
  window.matchMedia = mockMatchMedia;
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
  expect(mockMatchMedia).toHaveBeenCalledTimes(1);
  expect(mockMatchMedia).toHaveBeenCalledWith("(max-width: 768px)");
});

it("renders without map on server", () => {
  Meteor.isServer = true;
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
  Meteor.isServer = false;
});

it("updates search state and sets nav level on mount", () => {
  const mockDispatch = jest.fn();
  navActions.setLevel = jest.fn();
  const wrapper = shallow(generateComponent({
    dispatch: mockDispatch,
  }));
  expect(wrapper.state().showSearch).toBe(true);
  expect(mockDispatch).toHaveBeenCalledTimes(1);
  expect(navActions.setLevel).toHaveBeenCalledTimes(1);
  expect(navActions.setLevel).toHaveBeenCalledWith("BASIC_CONTENT");
});

it("onCardHover updates state from target", () => {
  const mockEvent = {
    currentTarget: {
      id: "test",
    },
  };
  const wrapper = shallow(generateComponent());
  wrapper.instance().onCardHover(mockEvent);
  expect(wrapper.state().hover).toBe("test");
});

it("onMarkerHover updates state from marker", () => {
  const wrapper = shallow(generateComponent());
  wrapper.instance().onMarkerHover({ id: "marker" });
  expect(wrapper.state().hover).toBe("marker");
});

it("getMarkers returns markers", () => {
  const wrapper = shallow(generateComponent());
  const result = wrapper.instance().getMarkers(defaultProps.data.groups.results);
  const group1 = defaultProps.data.groups.results[0];
  const group2 = defaultProps.data.groups.results[1];
  expect(result).toEqual([
    {
      latitude: group1.locations[0].location.latitude,
      longitude: group1.locations[0].location.longitude,
      id: group1.id,
    },
    {
      latitude: group2.locations[0].location.latitude,
      longitude: group2.locations[0].location.longitude,
      id: group2.id,
    },
  ]);
});

it("toggleSearch change show search state", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.state().showSearch).toBe(true);
  wrapper.instance().toggleSearch();
  expect(wrapper.state().showSearch).toBe(false);
  wrapper.instance().toggleSearch();
  expect(wrapper.state().showSearch).toBe(true);
});

it("toggleSearch change show search state", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.state().showTags).toBe(false);
  wrapper.instance().toggleTags();
  expect(wrapper.state().showTags).toBe(true);
  wrapper.instance().toggleTags();
  expect(wrapper.state().showTags).toBe(false);
});

it("removeQueryString calls preventDefault", () => {
  const wrapper = shallow(generateComponent());
  const mockPreventDefault = jest.fn();
  wrapper.instance().removeQueryString({
    preventDefault: mockPreventDefault,
  });
  expect(mockPreventDefault).toHaveBeenCalledTimes(1);
});

it("removeQueryString updates router with blank object", () => {
  const mockCreatePath = jest.fn().mockReturnValue("newPath");
  const mockReplace = jest.fn();
  const wrapper = shallow(generateComponent({
    router: {
      createPath: mockCreatePath,
      replace: mockReplace,
    },
  }));
  wrapper.instance().removeQueryString();
  expect(mockCreatePath).toHaveBeenCalledTimes(1);
  expect(mockCreatePath).toHaveBeenCalledWith({});
  expect(mockReplace).toHaveBeenCalledTimes(1);
  expect(mockReplace).toHaveBeenCalledWith("newPath");
});
