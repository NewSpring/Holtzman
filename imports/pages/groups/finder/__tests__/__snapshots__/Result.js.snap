// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with props 1`] = `
<div>
  <Connect(SplitContainerWithoutData)>
    <Right
      classes={
        Array [
          "background--left",
        ]
      }
      mobile={false}
    >
      <ControllableWrapper
        autoCenter={true}
        hover={null}
        markers={
          Array [
            Object {
              "id": "1",
              "latitude": 2,
              "longitude": 3,
            },
            Object {
              "id": "2",
              "latitude": 2,
              "longitude": 3,
            },
          ]
        }
        onChildClick={[Function]}
        onMarkerHover={[Function]}
      />
    </Right>
  </Connect(SplitContainerWithoutData)>
  <Left
    classes={
      Array [
        "background--light-secondary",
      ]
    }
    scroll={true}
  >
    <Layout
      LoadingComponent={[Function]}
      campusLocations={Array []}
      campuses={
        Array [
          "one",
          "two",
        ]
      }
      count={2}
      done={false}
      groups={
        Array [
          Object {
            "id": "1",
            "locations": Array [
              Object {
                "location": Object {
                  "latitude": 2,
                  "longitude": 3,
                },
              },
            ],
          },
          Object {
            "id": "2",
            "locations": Array [
              Object {
                "location": Object {
                  "latitude": 2,
                  "longitude": 3,
                },
              },
            ],
          },
        ]
      }
      loading={false}
      onCardHover={[Function]}
      query="test"
      removeQueryString={[Function]}
      schedules={
        Array [
          "1",
        ]
      }
      showSearch={true}
      showTags={false}
      tags={
        Array [
          "one",
          "two",
        ]
      }
      toggleSearch={[Function]}
      toggleTags={[Function]}
    />
  </Left>
</div>
`;

exports[`renders without map on mobile 1`] = `
<div>
  <Connect(SplitContainerWithoutData)>
    <Right
      classes={
        Array [
          "background--left",
        ]
      }
      mobile={false}
    />
  </Connect(SplitContainerWithoutData)>
  <Left
    classes={
      Array [
        "background--light-secondary",
      ]
    }
    scroll={true}
  >
    <Layout
      LoadingComponent={[Function]}
      campusLocations={Array []}
      campuses={
        Array [
          "one",
          "two",
        ]
      }
      count={2}
      done={false}
      groups={
        Array [
          Object {
            "id": "1",
            "locations": Array [
              Object {
                "location": Object {
                  "latitude": 2,
                  "longitude": 3,
                },
              },
            ],
          },
          Object {
            "id": "2",
            "locations": Array [
              Object {
                "location": Object {
                  "latitude": 2,
                  "longitude": 3,
                },
              },
            ],
          },
        ]
      }
      loading={false}
      onCardHover={[Function]}
      query="test"
      removeQueryString={[Function]}
      schedules={
        Array [
          "1",
        ]
      }
      showSearch={true}
      showTags={false}
      tags={
        Array [
          "one",
          "two",
        ]
      }
      toggleSearch={[Function]}
      toggleTags={[Function]}
    />
  </Left>
</div>
`;

exports[`renders without map on server 1`] = `
<div>
  <Connect(SplitContainerWithoutData)>
    <Right
      classes={
        Array [
          "background--left",
        ]
      }
      mobile={false}
    />
  </Connect(SplitContainerWithoutData)>
  <Left
    classes={
      Array [
        "background--light-secondary",
      ]
    }
    scroll={true}
  >
    <Layout
      LoadingComponent={[Function]}
      campusLocations={Array []}
      campuses={
        Array [
          "one",
          "two",
        ]
      }
      count={2}
      done={false}
      groups={
        Array [
          Object {
            "id": "1",
            "locations": Array [
              Object {
                "location": Object {
                  "latitude": 2,
                  "longitude": 3,
                },
              },
            ],
          },
          Object {
            "id": "2",
            "locations": Array [
              Object {
                "location": Object {
                  "latitude": 2,
                  "longitude": 3,
                },
              },
            ],
          },
        ]
      }
      loading={false}
      onCardHover={[Function]}
      query="test"
      removeQueryString={[Function]}
      schedules={
        Array [
          "1",
        ]
      }
      showSearch={true}
      showTags={false}
      tags={
        Array [
          "one",
          "two",
        ]
      }
      toggleSearch={[Function]}
      toggleTags={[Function]}
    />
  </Left>
</div>
`;
